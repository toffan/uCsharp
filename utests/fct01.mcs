int max(int a, int b) {
    if (a > b) {
        return a;
    } else {
        return b;
    }
}

int main() {
    int c = max(5, 9);
    return 0;
}
