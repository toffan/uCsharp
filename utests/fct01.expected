JUMP __max
_max:
    LOAD (1) -2[LB]
    LOAD (1) -1[LB]
    SUBR IGtr
    SUBR I2B
    JUMPIF (0) ELSE_0
    LOAD (1) -2[LB]
    RETURN (1) 2
    POP (0) 0
    JUMP ENDIF_0
ELSE_0:
    LOAD (1) -1[LB]
    RETURN (1) 2
    POP (0) 0
ENDIF_0:
    RETURN (0) 2
__max:

JUMP __main
_main:
    LOADL 5
    LOADL 9
    CALL (SB) _max

    LOADL 0
    RETURN (1) 0
    RETURN (0) 0
__main:
    CALL (SB) _main
    HALT
